Go docs update (#130)
undefinedlabs/scope-docs: docs/go-sql-instrumentation.md

---
id: go-sql-instrumentation
title: Scope Go Agent SQL Instrumentation
sidebar_label: SQL Instrumentation
---

## Instrumenting the SQL driver

First, make sure the agent has been [installed](go-installation.md) in your Go application.

The Scope Go agent provides a wrapper function to instrument an existing `database/sql` driver. 

Currently, the instrumentation supports the following libraries:

- Postgres (https://github.com/lib/pq)
- MySql (https://github.com/go-sql-driver/mysql)

To instrument the driver you need to register a new one using the wrapper:

```go
import (
    "database/sql"
    
    "github.com/lib/pq"
    "github.com/go-sql-driver/mysql"

    scopesql "go.undefinedlabs.com/scopeagent/instrumentation/sql"
)

func main() {
    // ...
	

> 	sql.Register("scope-postgres", scopesql.WrapDriver(&pq.Driver{}))

Reviewer: ```suggestion
    sql.Register("scope-postgres", scopesql.WrapDriver(&pq.Driver{}))
```

    sql.Register("scope-mysql", scopesql.WrapDriver(&mysql.MySQLDriver{}))

    // ...
}
```

> Also, by using `scopesql.WithStatements()` the sql statement values will be instrumented aswell. 

Reviewer: ```suggestion
Also, by using `scopesql.WithStatements()` the SQL statement values will be instrumented as well. 
```

And I would separate this into a different section, as it's not part of the basic installation. And also, statements are always captured right? The option is to enable statement _values_. The option has to be renamed.

Author: done, please check again.

    // ...
}
```

After this, in order for the Scope Go agent to trace queries to the database, you have to open the instrumented driver and attach the current context in each query. 
For example:

<!--DOCUSAURUS_CODE_TABS-->
<!--Postgres-->

```go
import (
	"database/sql"

    _ "github.com/lib/pq"
    
	"go.undefinedlabs.com/scopeagent"
	scopesql "go.undefinedlabs.com/scopeagent/instrumentation/sql"
)

func main() {
    // ...
    
	psqlInfo := fmt.Sprintf("host=%s port=%d user=%s "+"password=%s dbname=%s sslmode=disable",
		host, port, user, password, dbname)

	db, err := sql.Open("scope-postgres", psqlInfo)
	if err != nil {
		panic(err)
	}
	defer db.Close()

	selectQuery := "SELECT * FROM my_table"
	rows, err := db.QueryContext(ctx, selectQuery)
	if err != nil {
		panic(err)
	}
	defer rows.Close()

    // ...
}
```

<!--MySql-->

```go
import (
	"database/sql"

    _ "github.com/go-sql-driver/mysql"
    
	"go.undefinedlabs.com/scopeagent"
	scopesql "go.undefinedlabs.com/scopeagent/instrumentation/sql"
)

func main() {
    // ...

	db, err := sql.Open("scope-mysql", "user:password@/dbname")
	if err != nil {
		panic(err)

> 	}

Reviewer: You are mixing here tabs and spaces. Check that all the code snippets are consistent.

Author: ok

---
id: go-sql-instrumentation
title: Scope Go Agent SQL Instrumentation
sidebar_label: SQL Instrumentation
---

## Instrumenting the SQL driver

First, make sure the agent has been [installed](go-installation.md) in your Go application.

The Scope Go agent provides a wrapper function to instrument an existing `database/sql` driver. 

Currently, the instrumentation supports the following libraries:

- Postgres (https://github.com/lib/pq)

> - MySql (https://github.com/go-sql-driver/mysql)

Reviewer: ```suggestion
- MySQL (https://github.com/go-sql-driver/mysql)
```

    "github.com/lib/pq"
    "github.com/go-sql-driver/mysql"

    scopesql "go.undefinedlabs.com/scopeagent/instrumentation/sql"
)

func main() {
    // ...

    sql.Register("scope-postgres", scopesql.WrapDriver(&pq.Driver{}))
    sql.Register("scope-mysql", scopesql.WrapDriver(&mysql.MySQLDriver{}))

    // ...
}
```

After this, in order for the Scope Go agent to trace queries to the database, you have to open the instrumented driver and attach the current context in each query. 
For example:

<!--DOCUSAURUS_CODE_TABS-->
<!--Postgres-->

```go
import (
    "database/sql"

    _ "github.com/lib/pq"

    "go.undefinedlabs.com/scopeagent"
    scopesql "go.undefinedlabs.com/scopeagent/instrumentation/sql"
)

func main() {
    // ...

    psqlInfo := fmt.Sprintf("host=%s port=%d user=%s "+"password=%s dbname=%s sslmode=disable",
        host, port, user, password, dbname)

    db, err := sql.Open("scope-postgres", psqlInfo)
    if err != nil {
        panic(err)
    }
    defer db.Close()

    selectQuery := "SELECT * FROM my_table"
    rows, err := db.QueryContext(ctx, selectQuery)
    if err != nil {
        panic(err)
    }
    defer rows.Close()

    // ...
}
```

> <!--MySql-->

Reviewer: ```suggestion
<!--MySQL-->
```
